dvc/remote/base.py

```diff
@@ -288,16 +294,71 @@ def changed_cache(self, checksum):
 
         return False
 
+    def _parallel(self, func, args):
+        """Wraps iterator into threadsafe_iterator and executes func in JOBS
+        threads in parallel
+        """
+        results = []
+        args = threadsafe_iterator(args)
+        with ThreadPoolExecutor(max_workers=self.JOBS) as executor:
+            futures = [executor.submit(func, args) for i in range(self.JOBS)]
+            for future in concurrent.futures.as_completed(futures):
+                try:
+                    data = future.result()
+                except Exception:
```

Comment:
seems not the right way to handle this. What kind of errors do we except? Could we actually ignore them and return a meaningful result?

Reply:
especially considering that logic with iterator - we'll be ignoring a big chuck of the result potentially